scripts/update-registry.ts

```
@@ -0,0 +1,276 @@
+  }
+
+  // Read all JSON files in the version directory
+  const files = await fs.readdir(assetsDir);
+  const jsonFiles = files.filter((file) => file.endsWith('.json'));
+
+  if (jsonFiles.length === 0) {
+    throw new Error(`No JSON files found in ${assetsDir}`);
+  }
+
+  debug(`Found ${jsonFiles.length} JSON files to update`);
+
+  // Pre-check: Validate all contracts and check if deployment is already fully supported
+  const deploymentType = options.deploymentType as AddressType;
+  const contractsToProcess: Array<{ file: string; path: string; json: SingletonDeploymentJSON }> = [];
+  const alreadySupported: string[] = [];
+  const missingDeploymentType: string[] = [];
+
+  for (const file of jsonFiles) {
+    const filePath = path.join(assetsDir, file);
+    const content = await fs.readFile(filePath, 'utf-8');
+    const json: SingletonDeploymentJSON = JSON.parse(content);
+
+    // Check if deployment type exists for this contract
+    if (!json.deployments[deploymentType as keyof typeof json.deployments]) {
+      missingDeploymentType.push(json.contractName);
+      continue;
+    }
+
+    // Check if chain ID already exists with this deployment type
+    const existingValue = json.networkAddresses[options.chainId];
+    if (existingValue !== undefined) {
+      const isArray = Array.isArray(existingValue);
+      const hasDeploymentType = isArray
+        ? existingValue.includes(deploymentType)
+        : existingValue === deploymentType;
+
+      if (hasDeploymentType) {
+        alreadySupported.push(json.contractName);
+        continue;
+      }
+    }
+
+    // Contract needs to be updated
+    contractsToProcess.push({ file, path: filePath, json });
+  }
+
+  // Report pre-check results
+  if (missingDeploymentType.length > 0) {
+    console.warn(
+      `⚠️  ${missingDeploymentType.length} contract(s) don't support deployment type "${deploymentType}": ${missingDeploymentType.join(', ')}`
+    );
```

I think you should error out here instead of continuing with a partial application. Our CI doesn't allow these partial states anyway (where we add a deployment to some files but not others).